build.c

```diff
@@ -599,17 +645,25 @@ build(void)
 		}
 		if (numjobs == 0)
 			break;
-		if (poll(fds, jobslen, 5000) < 0)
+		if (poll(fds, jobslen, 5000) < 0 && errno != EINTR)
 			fatal("poll:");
 		for (i = 0; i < jobslen; ++i) {
-			if (!fds[i].revents || jobwork(&jobs[i]))
+			/* active job? */
+			if (!jobs[i].edge)
 				continue;
```

Comment:
That would be a job with `fds[i].fd==-1`, there are obviously many since the fds and jobs array are more like arenas for a linked list. Normally they are skipped because a negative file descriptor gives revents == 0, but killall has to find a different way to skip the inactive jobs, because it has to operate even on those that have no input available on the file descriptor.

I will switch to checking `fds[i].fd==-1`, but clearing `j->edge` in jobdone is probably a good idea regardless, as a NULL pointer dereference is better than doing random stuff on the edge on the second (wrong) completion.

Reply:
I will add a note in the commit message